src/EmbeddedManifold.jl

```diff
@@ -89,6 +89,16 @@ function EmbeddedManifold(
     return EmbeddedManifold{𝔽,MT,NT,ET}(M, N)
 end
 
+function allocate_result(M::AbstractEmbeddedManifold, f::typeof(embed), x...)
+    T = allocate_result_type(M, f, x)
+    return allocate(x[end], T, representation_size(decorated_manifold(M)))
```

Comment:
Why the last and not the first?

Reply:
In `x` is either point and we want to return point, or point and vector and we want to return vector, so the result is more similar to the last argument.

Reply:
Thanks for the explanation – clever!